spec/initializers/rack/attack_spec.rb

```diff
@@ -11,12 +11,11 @@
   describe "search_throttle" do
     it "throttles /search endpoints based on IP" do
       Timecop.freeze do
-        allow(Search::User).to receive(:search_documents).and_return({})
         valid_responses = Array.new(5).map do
-          get "/search/users", headers: { "HTTP_FASTLY_CLIENT_IP" => "5.6.7.8" }
+          get "/search/usernames", headers: { "HTTP_FASTLY_CLIENT_IP" => "5.6.7.8" }
```

Comment:
Took me a moment to see that /search/users was being removed from config/routes.rb (and User from Search) and that the (already existing) /search/usernames is being used instead. I thought this could have been an unrelated fix at first glance.

Does this test now _actually_ run the search request (which had been stubbed out for the ES search by returning an empty hash) - or is the absence of an argument sufficient to avoid making the query?


```
  def usernames
    result = Search::Postgres::Username.search_documents(params[:username])

    render json: { result: result }
  end
```

My assumption is that params[:username] is empty here in the test, the search service doesn't exit early on `term.empty?` and instead calls the models search method

```
  scope :search_by_name_and_username, lambda { |term|
    where(
      sanitize_sql_array(
        [
          "to_tsvector('simple', coalesce(name::text, '')) @@ to_tsquery('simple', ? || ':*')",
          connection.quote(term),
        ],
      ),
    ).or(
      where(
        sanitize_sql_array(
          [
            "to_tsvector('simple', coalesce(username::text, '')) @@ to_tsquery('simple', ? || ':*')",
            connection.quote(term),
          ],
        ),
      ),
    )
  }
```

It looks like we _will_ be running the query code multiple times here to trigger the rate limit?

Reply:
Great call, updated to stub it!